src/test/java/uk/gov/hmcts/reform/sscs/ccd/deserialisation/SscsCaseCallbackDeserializerTest.java

```diff
@@ -242,4 +245,18 @@ public void should_deserialize_update_further_evidence_callback() throws IOExcep
         assertEquals("http://dm-store:4506/documents/5f574d09-1590-446e-bc02-1f2437688390", directionResponse.getDirectionResponses().get(0).getValue().getDocumentLink().getDocumentUrl());
         assertEquals("http://dm-store:4506/documents/5f574d09-1590-446e-bc02-1f2437688390/binary", directionResponse.getDirectionResponses().get(0).getValue().getDocumentLink().getDocumentBinaryUrl());
     }
+
+    @Test
+    @Parameters({"adminAppealWithdrawnCallback.json", "updateFurtherEvidence.json"})
+    public void should_deserialise_and_serialise(final String jsonFileName) throws IOException, JSONException {
+        sscsCaseCallbackDeserializer = new SscsCaseCallbackDeserializer(mapper);
+
+        String path = getClass().getClassLoader().getResource(jsonFileName).getFile();
+        String json = FileUtils.readFileToString(new File(path), StandardCharsets.UTF_8.name());
+        Callback<SscsCaseData> actualSscsCaseCallback = sscsCaseCallbackDeserializer.deserialize(json);
+
+        String valueAsString = mapper.writeValueAsString(actualSscsCaseCallback);
+
+        JSONAssert.assertEquals(valueAsString, json,  JSONCompareMode.LENIENT);
```

Comment:
had some issues using the library to compare the Json string because it's too good. I had to swap expected and actual as the actual is a subset of the expected because the Json file is from a ccd response rather than a Callback<SSCSCaseData> if that makes sense.